src/utils/arweave.ts

```diff
@@ -54,3 +57,66 @@ export const paginationParamsToTags = <T>(
 
   return pruneTags(tags);
 };
+
+/**
+ * Get the epoch with distribution data for the current epoch
+ * @param arweave - The Arweave instance
+ * @returns The epoch with distribution data
+ */
+export const getEpochDataFromGql = async ({
+  arweave,
+  epochIndex,
+  processId = ARIO_TESTNET_PROCESS_ID,
+}: {
+  arweave: Arweave;
+  epochIndex: number;
+  processId?: string;
+}): Promise<AoEpochData | undefined> => {
+  // fetch from gql
+  const query = epochDistributionNoticeGqlQuery({ epochIndex, processId });
+  const response = await arweave.api.post('graphql', query);
+  // parse the nodes to get the id
+  if (response.data.data.transactions?.edges?.length === 0) {
+    return undefined;
+  }
+  const id = response.data.data.transactions.edges[0].node.id;
+  // fetch the transaction from arweave
+  const transaction = await arweave.api.get<AoEpochData>(id);
+  const data = transaction.data;
+  // assert it is the correct type
+  return parseAoEpochData(data);
+};
+
+export const epochDistributionNoticeGqlQuery = ({
+  epochIndex,
+  processId = ARIO_TESTNET_PROCESS_ID,
+}: {
+  epochIndex: number;
+  processId?: string;
+}): string => {
+  // write the query
+  const gqlQuery = JSON.stringify({
+    query: `
+      query {
+        transactions(
+          tags: [
+            { name: "From-Process", values: ["${processId}"] }
+            { name: "Action", values: ["Epoch-Distribution-Notice"] }
+            { name: "Epoch-Index", values: ["${epochIndex}"] }
+            { name: "Data-Protocol", values: ["ao"] }
+          ],
+          owners: ["fcoN_xJeisVsPXA-trzVAuIiqO3ydLQxM-L4XbrQKzY"],
```

Comment:
nit: bring out to a constant or as an authorities input

Reply:
will leave as is for now, we don't have any known trusted authorities. don't want to leave room for potential test addresses that return unexpected data. can revisit post AO mainnet